pilot/cmd/pilot-agent/main.go

```diff
@@ -97,26 +98,27 @@ var (
 				}
 			}
 
-			// set values from registry platform
-			if len(role.IPAddress) == 0 {
-				if registry == serviceregistry.KubernetesRegistry {
```

Comment:
This change the original behavior.

Reply:
Yes. But it’s good isn’t it? We can use a uniform way for all platforms